packages/select/src/components/query-list/queryList.tsx

```diff
@@ -148,16 +149,6 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
             // reset the flag
             this.shouldCheckActiveItemInViewport = false;
         }
-
-        // reset active item (in the same step) if it's no longer valid
-        // Also don't fire the event if the active item is already undefined and there is nothing to pick
-        const activeIndex = this.getActiveIndex();
-        if (
-            this.props.activeItem !== undefined &&
-            (activeIndex < 0 || isItemDisabled(this.props.activeItem, activeIndex, this.props.itemDisabled))
-        ) {
-            this.setFirstActiveItem();
-        }
```

Comment:
this was the bad logic. i think the `setState` callback above was the main culprit. refactored that to simply reuse the `setQuery` logic, and merged this logic into the reset check in `setQuery` below.